### ISSUE TITLE
"..."u8 doesn't work in attributes

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#60626]
**Version Used**: 
4.3.0-1.22206.2 (91e45f95)

**Steps to Reproduce**:
```C#
[SomeAttributeThatAcceptsByteArray("text"u8)]
``` 

**Expected Behavior**:
Compiles successfully, with:
```C#
[SomeAttributeThatAcceptsByteArray("text"u8)]
```
becoming the equivalent of:
```C#
[SomeAttributeThatAcceptsByteArray(new byte[] { (byte)'t', (byte)'e', (byte)'x', (byte)'t' })]
```

**Actual Behavior**:
Fails to compile with:
```
error CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type
```

**Comment by jcouv** [comment]
@AlekseyTs to triage and advise

**Comment by AlekseyTs** [comment]
The behavior is according to the current design. The error message is clearly explains the reason. The `U8` literal is not "a constant expression, typeof expression or array creation expression". If we want to support this scenario, requirements for attribute arguments should be extended in the language.

**Comment by stephentoub** [comment]
> The U8 literal is not "a constant expression, typeof expression or array creation expression"

It's logically an array creation expression, so I don't think it's clear from the error message.

But regardless, I think we should enable this.

**Comment by AlekseyTs** [comment]
> It's logically an array creation expression, so I don't think it's clear from the error message.

To be precise, an "array creation expression" refers to a specific syntactic language construct rather than to something that one might consider to be logically equivalent. And `U8` literal is not an "array creation expression". If I remember correctly, supporting this scenario was not part of the proposal. Nevertheless, I have no objection against adjusting the language rules to enable this scenario. We should discuss that at LDM.

**Comment by stephentoub** [comment]
> To be precise, an "array creation expression" refers to a specific syntactic language construct rather than to something that one might consider to be logically equivalent. And U8 literal is not an "array creation expression".

Understood.  I was responding to the "clearly explains" part.  Clarity is in the eye of the beholder.

**Comment by Youssef1313** [comment]
If a change is taken here, the analyzer implemented #60647 will need changes.

**Comment by AlekseyTs** [comment]
We are changing natural type on `u8` literals to ```ReadOnlySpan<byte>```, which has no conversion to ```byte[]```.  Also, at the moment, ```ReadOnlySpan<byte>``` is not supported as an attribute argument type and metadata specification doesn't define how  a value for an argument like that could be encoded.
